### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by DmitryBarskov**

### CODE REVIEW

File: app/models/jwt_codec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# Encodes/decodes JSON Web Tokens
+class JwtCodec
```
Comment (EgorBarakhov) [inline app/models/jwt_codec.rb:4]: Are you sure this is the model, not service?

File: app/models/session.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    JwtCodec.new.encode({ id: user.id })
+  end
+
+  private
+
+  attr_reader :email, :password
+
+  def user
+    @user ||= User.find_by(email: @email)&.authenticate(@password)
+  end
```
Comment (EgorBarakhov) [inline app/models/session.rb:27]: Don't you want to create separate PORO class to make this method look like
```suggestion
  def user
    @user ||= Users::Authenticate(email: @email, password: @password)
  end
```

File: app/models/session.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  def authenticated
+    errors.add :base, :authenticated if user.blank?
```
Comment (EgorBarakhov) [inline app/models/session.rb:30]: Why don't you create a separate validator class?

File: config/routes.rb [hunk 4]
```diff
@@ -4,6 +4,7 @@
   namespace :api do
     namespace :v1 do
       resources :users
+      resources :sessions
```
Comment (EgorBarakhov) [inline config/routes.rb:7]:
```suggestion
      resources :sessions, only: %i[create]
```

**Review by EgorBarakhov — CHANGES_REQUESTED** [review]

File: app/models/jwt_codec.rb (hunk 1 above)
Comment (DmitryBarskov) [reply app/models/jwt_codec.rb:4]: There is no such thing as service in MVC :)

File: app/models/session.rb (hunk 3 above)
Comment (DmitryBarskov) [reply app/models/session.rb:30]: IMO it's an overkill in this case.

File: app/models/session.rb (hunk 2 above)
Comment (DmitryBarskov) [reply app/models/session.rb:27]: Thanks, I like it. Will apply your suggestion in a future PR.

File: app/models/session.rb (hunk 3 above)
Comment (DmitryBarskov) [reply app/models/session.rb:30]: Anyway, `Session` is to be refactored in #9 to be testable. I moved this logic to another class and I will apply your suggestion to the other class. Thanks

File: app/models/jwt_codec.rb (hunk 1 above)
Comment (DmitryBarskov) [reply app/models/jwt_codec.rb:4]: Seriously, I want to build an object-oriented design in this project so I rejected idea of "service layer" (interactors, service objects, etc). They are just disguised procedures.

**Pull request merged**